content-security-policy/support/testharness-helper.sub.js

```diff
@@ -0,0 +1,81 @@
+const Host = {
+  SAME_ORIGIN: "same-origin",
+  CROSS_ORIGIN: "cross-origin",
+};
+
+const PolicyHeader = {
+  CSP: "echo-policy.py?policy=",
+  EMBEDDING_CSP: "echo-embedding-csp.py",
+  ALLOW_CSP_FROM: "echo-allow-csp-from.py",
+} 
+
+const IframeLoad = {
+  EXPECT_BLOCK: true,
+  EXPECT_LOAD: false,
+}
+
+function generateURL(host, path) {
+  var url = new URL("http://{{host}}:{{ports[http][0]}}/content-security-policy/support/");
+  url.hostname = host == Host.SAME_ORIGIN ? "{{host}}" : "{{domains[天気の良い日]}}";
+  url.pathname += path;
+
+  return url.toString();
+}
+
+function getOrigin() {
+  var url = new URL("http://{{host}}:{{ports[http][0]}}/");
+  return url.toString();
+}
+
+function generateUrlWithAllowCSPFrom(host, allowCspFrom) {
+    var url = generateURL(host, PolicyHeader.ALLOW_CSP_FROM);
+    return url + "?allow_csp_from=" + allowCspFrom;
+}
+
+function assert_iframe_with_csp(t, url, csp, shouldBlock, urlId, blockedURI) {
+  var i = document.createElement('iframe');
+  i.src = url + "&id=" + urlId;
+  i.csp = csp;
+
+  var loaded = {};
+  window.addEventListener("message", function (e) {
+    if (e.source != i.contentWindow)
+        return;
+    if (e.data["loaded"])
+        loaded[e.data["id"]] = true;
+  });
+
+  if (shouldBlock) {
+    // Assert iframe does not load.
+    window.onmessage = function (e) {
+      if (e.source != i.contentWindow)
+          return;
+      t.unreached_func('No message should be sent from the frame.');
+    }
+    i.onload = t.step_func(function () {
+      // Delay the check until the postMessage has a chance to execute.
+      setTimeout(t.step_func_done(function () {
+           assert_equals(loaded[urlId], undefined);
+      }), 1);
+    });
+  } else if (blockedURI) {
+    // Assert iframe loads with an expected violation.
+    window.addEventListener('message', t.step_func(e => {
+      if (e.source != i.contentWindow || e.data["securitypolicyviolation"] != true)
```

Comment:
Shouldn't the test fail if `securitypolicyviolation` is `false` and `blockedURI` isn't?